### PR TITLE
Allow sending of previous shipfile emails

### PR DESCRIPTION
**Pull request by lewisrenfrew**

### CODE REVIEW

File: tests/Integration/Service.Tests/WandModuleSpecs/WhenAddingShipfile.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        }
+
+        [Test]
+        public void ShouldCallFacadeService()
+        {
+            this.ShipfileService.Received().Add(Arg.Any<ConsignmentShipfileResource>());
+        }
+
+        [Test]
+        public void ShouldReturnDeleted()
```
Comment (richardiphillips) [inline tests/Integration/Service.Tests/WandModuleSpecs/WhenAddingShipfile.cs:60]: deleted?

File: src/Facade/Services/ConsignmentShipfileFacadeService.cs [hunk 2]
```diff
@@ -71,5 +76,31 @@ public IResult<ConsignmentShipfile> DeleteShipfile(int id)
...
+
+            if (consignment == null)
+            {
+                return new BadRequestResult<ConsignmentShipfile>("Consignment Not Found");
+            }
+
+            var shipfile = new ConsignmentShipfile
+                               {
+                                   ConsignmentId = consignment.ConsignmentId,
+                                   Id = this.repository.FindAll().Max(x => x.Id) + 1
```
Comment (iaincrawf) [inline src/Facade/Services/ConsignmentShipfileFacadeService.cs:98]: do we need to specify an id for shipfiles? I haven't seen us have to do this ^ before, just seen us setting an Id for ones that use the get_next_val function or whatever (doesn't mean we don't need to here though)

**Review by iaincrawf — APPROVED** [review]

File: src/Facade/Services/ConsignmentShipfileFacadeService.cs (hunk 2 above)
Comment (richardiphillips) [inline src/Facade/Services/ConsignmentShipfileFacadeService.cs:98]: i think this should be done in the repo

File: tests/Unit/Facade.Tests/ConsignmentShipfilesFacadeServiceTests/WhenAdding.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    public class WhenAdding: ContextBase
+    {
+        private IResult<ConsignmentShipfile> result;
+
+        [SetUp]
+        public void SetUp()
+        {
+            this.ConsignmentRepository.FindBy(Arg.Any<Expression<Func<Consignment, bool>>>())
+                .Returns(new Consignment {});
+            this.Repository.FindAll().Returns(new List<ConsignmentShipfile> { new ConsignmentShipfile { Id = 1 } }.AsQueryable());
```
Comment (richardiphillips) [inline tests/Unit/Facade.Tests/ConsignmentShipfilesFacadeServiceTests/WhenAdding.cs:28]: is this findall used?

**Review by richardiphillips — APPROVED** [review]

File: src/Facade/Services/ConsignmentShipfileFacadeService.cs (hunk 2 above)
Comment (richardiphillips) [reply src/Facade/Services/ConsignmentShipfileFacadeService.cs:98]: with a sequence would be better

File: tests/Unit/Facade.Tests/ConsignmentShipfilesFacadeServiceTests/WhenAdding.cs (hunk 3 above)
Comment (lewisrenfrew) [reply tests/Unit/Facade.Tests/ConsignmentShipfilesFacadeServiceTests/WhenAdding.cs:28]: It is for generating the Id, but you're previous comment to just use a sequence is a better idea anyway

File: src/Facade/Services/ConsignmentShipfileFacadeService.cs (hunk 2 above)
Comment (lewisrenfrew) [reply src/Facade/Services/ConsignmentShipfileFacadeService.cs:98]: Using a sequence is a better idea anyway

**Pull request merged**